airbyte-integrations/connectors/destination-timeplus/destination_timeplus/destination.py

```diff
@@ -0,0 +1,122 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+
+from typing import Any, Iterable, Mapping
+
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.destinations import Destination
+from airbyte_cdk.models import AirbyteConnectionStatus, AirbyteMessage, ConfiguredAirbyteCatalog, Status, DestinationSyncMode, Type, AirbyteStream
+
+import json
+from timeplus import Stream, Environment
+
+class DestinationTimeplus(Destination):
+    def write(
+        self, config: Mapping[str, Any], configured_catalog: ConfiguredAirbyteCatalog, input_messages: Iterable[AirbyteMessage]
+    ) -> Iterable[AirbyteMessage]:
+        """
+        Reads the input stream of messages, config, and catalog to write data to the destination.
+
+        This method returns an iterable (typically a generator of AirbyteMessages via yield) containing state messages received
+        in the input message stream. Outputting a state message means that every AirbyteRecordMessage which came before it has been
+        successfully persisted to the destination. This is used to ensure fault tolerance in the case that a sync fails before fully completing,
+        then the source is given the last state message output from this method as the starting point of the next sync.
+
+        :param config: dict of JSON configuration matching the configuration declared in spec.json
+        :param configured_catalog: The Configured Catalog describing the schema of the data being received and how it should be persisted in the
+                                    destination
+        :param input_messages: The stream of input messages received from the source
+        :return: Iterable of AirbyteStateMessages wrapped in AirbyteMessage structs
+        """
+        endpoint=config["endpoint"]
+        apikey=config["apikey"]
+        if endpoint[-1]=='/':
+            endpoint=endpoint[0:len(endpoint)-1]
+        env = Environment().address(endpoint).apikey(apikey)
+        stream_list = Stream(env=env).list()
+        all_streams = {s.name for s in stream_list}
+
+        # only support "overwrite", "append"
+        for configured_stream in configured_catalog.streams:
+            is_overwrite = configured_stream.destination_sync_mode == DestinationSyncMode.overwrite
+            stream_exists = configured_stream.stream.name in all_streams
+            if is_overwrite and stream_exists:
+                # delete the existing stream
+                Stream(env=env).name(configured_stream.stream.name).get().delete()
+            if is_overwrite or not stream_exists:
+                # create a new stream
+                DestinationTimeplus.create_stream(env, configured_stream.stream)
```

Comment:
```suggestion
            if is_overwrite: 
                if stream_exists:
                    # delete the existing stream
                    Stream(env=env).name(configured_stream.stream.name).get().delete()
                else: 
                       # create a new stream
                       DestinationTimeplus.create_stream(env, configured_stream.stream)
          else:
               raise Error("Sync mode not implemented)
```

Reply:
It's more complicated than that.  We do support is_overwrite=False

All four conditions
<img width="376" alt="image" src="https://user-images.githubusercontent.com/5076438/222022669-a070a6e8-ab64-4c6c-af40-126de69acb54.png">

My previous commits to minimize the duplicated code, but yes, it's hard to maintain. I will rewrite this part.

Reply:
pls review the new code.